File: urbanairship-layout/src/main/java/com/urbanairship/android/layout/property/MediaType.java
/* Copyright Airship and Contributors */

package com.urbanairship.android.layout.property;

import com.urbanairship.json.JsonException;

import java.util.Locale;

import androidx.annotation.NonNull;

public enum MediaType {
    IMAGE("image"),
    VIDEO("video"),
    YOUTUBE("youtube"),
    VIMEO("vimeo");

    @NonNull
    private final String value;

    MediaType(@NonNull String value) {
        this.value = value;
    }

    @NonNull
    public static MediaType from(@NonNull String value) throws JsonException {
        for (MediaType type : MediaType.values()) {
            if (type.value.equals(value.toLowerCase(Locale.ROOT))) {
                return type;
            }
        }
        throw new JsonException("Unknown MediaType value: " + value);
    }

    @NonNull
    @Override
    public String toString() {
        return name().toLowerCase(Locale.ROOT);
    }
}

File: urbanairship-layout/src/main/java/com/urbanairship/android/layout/property/MediaType.kt
/* Copyright Airship and Contributors */
package com.urbanairship.android.layout.property

import com.urbanairship.json.JsonException
import com.urbanairship.json.JsonValue

public enum class MediaType(
    private val value: String,
    public val isPlayable: Boolean
) {
    IMAGE("image", isPlayable = false),
    VIDEO("video", isPlayable = true),
    YOUTUBE("youtube", isPlayable = true),
    VIMEO("vimeo", isPlayable = true);

    override fun toString(): String {
        return name.lowercase()
    }

    public companion object {

        @Throws(JsonException::class)
        public fun from(value: JsonValue): MediaType {
            val content = value.requireString().lowercase()
            return entries.firstOrNull { it.value == content }
                ?: throw JsonException("Unknown MediaType value: $content")
        }
    }
}
